tree/dataframe/inc/ROOT/RDFInterface.hxx

```diff
@@ -1419,6 +1419,26 @@ public:
    /// This is not an action nor a transformation, just a query to the RDataFrame object.
    std::vector<std::string> GetFilterNames() { return RDFInternal::GetFilterNames(fProxiedPtr); }
 
+   /// \brief Returns the names of the defined columns
+   ///
+   /// This is not an action nor a transformation, just a simple utility to
+   /// get the columns names that have been defined up to the node.
+   /// If no custom column has been defined, e.g. on a root node, it returns an
+   /// empty array.
+   ColumnNames_t GetDefinedColumnNames()
+   {
+      ColumnNames_t definedColumns;
+
+      auto columns = fCustomColumns.GetColumns();
+
+      for(auto column: columns){
+         if (!RDFInternal::IsInternalColumn(column.first) && !column.second->IsDataSourceColumn())
+            definedColumns.emplace_back(column.first);
+      }
```

Comment:
Perhaps it's worth investing in a std::copy_if based solution?

Reply:
Thanks Danilo. `std::copy_if` is not suitable approaches because I am copying one element of the map into an array, not the whole map entry. 
`std::transform` can't be used because it can't filter entries.